### PR TITLE
serde support for Record objects

### PR DESCRIPTION
**Pull request by pmarks**
optional support for using Record objects with serde. Useful for integrating BAM data with Rust serde infrastructure.

### CODE REVIEW

File: src/bam/record.rs [hunk 1]
```diff
@@ -69,6 +69,56 @@ impl Clone for Record {
     }
 }
 
+impl PartialEq for Record {
+    fn eq(&self, other: &Record) -> bool {
```
Comment (johanneskoester) [inline src/bam/record.rs]: I think I would prefer a big lazy OR here: ` self.tid() != other.tid() || self.pos() != other.pos() || ...`

File: src/bam/record.rs [hunk 2]
```diff
@@ -414,12 +481,8 @@ impl Record {
...
             }
         };
-
-        if ret < 0 {
-            Err(AuxWriteError::Some)
-        } else {
-            Ok(())
-        }
+        
+        Ok(())
```
Comment (johanneskoester) [inline src/bam/record.rs]: Why removing the error handling here?

File: src/bam/record.rs (hunk 1 above)
Comment (pmarks) [reply src/bam/record.rs]: good call

File: src/bam/record.rs (hunk 2 above)
Comment (pmarks) [reply src/bam/record.rs]: restored the error handling. In htslib 1.7, aux_append returns void, so we're going to need to drop this eventually.

**Pull request merged**